modules/http4s/test/src-ce3/smithy4s/compliancetests/ProtocolComplianceTest.scala

```diff
@@ -131,22 +150,69 @@ object ProtocolComplianceTest extends EffectSuite[IO] with BaseCatsSuite {
     codecApi
       .decodeFromByteArray[Document](codec, bytes)
       .getOrElse(sys.error("unable to decode smithy model into document"))
+
   }
 
-  private def runInWeaver(tc: ComplianceTest[IO]): IO[TestOutcome] = Test(
-    tc.name,
-    tc.run
-      .map[Expectations] {
-        case Left(value) =>
-          Expectations.Helpers.failure(value)
-        case Right(_) =>
-          Expectations.Helpers.success
-      }
-      .attempt
-      .map {
-        case Right(expectations) => expectations
-        case Left(e) => weaver.Expectations.Helpers.failure(e.getMessage)
+  private def runInWeaver(
+      allowList: AllowRules,
+      tc: ComplianceTest[IO]
+  ): IO[TestOutcome] = {
+    val runner: IO[Expectations] = {
+      if (allowList.isAllowed(tc) || tc.show.contains("alloy")) {
+        tc.run
+          .map(res => expectSuccess(res))
+          .attempt
+          .map {
+            case Right(expectations) => expectations
+            case Left(throwable) =>
+              Expectations.Helpers.failure(
+                s"unexpected error when running test ${throwable.getMessage}"
+              )
+          }
+      } else tc.run.map(res => expectFailure(res))
+    }
+
+    Test(
+      tc.show,
+      runner
+    )
+  }
+
+  def expectSuccess(
+      res: ComplianceTest.ComplianceResult
+  ): Expectations = {
+    res.bifoldMap(
+      Expectations.Helpers.failure(_),
+      _ => Expectations.Helpers.success
+    )
+  }
+
+  def expectFailure(
+      res: ComplianceTest.ComplianceResult
+  ): Expectations = {
+    res.bifoldMap(
+      _ => Expectations.Helpers.success,
+      _ =>
+        Expectations.Helpers.failure(
+          "An expected failure has passed, we might need to update the AllowList  "
+        )
+    )
+  }
+
+  def readConfigFromJar: IO[Seq[Byte]] = {
+    Resource
+      .fromAutoCloseable(
+        IO(
+          new java.io.BufferedReader(
+            new java.io.InputStreamReader(
+              getClass.getResourceAsStream("/META-INF/json/test-config.json")
```

Comment:
That method is JVM specific : you'll have to dump it onto the file system at build time.

Alternatively, we could put this configuration into a smithy-compliant json file as metadata, so that it'd already benefit from the build logic you added to dump the model as json file. 

We'd then add a `def metadata: Map[String, Document]` to our `DynamicSchemaIndex`, that could be really useful in the future.

Reply:
yes , I am aware , thats why its still in draft , i'm just trying to dot a couple i's before tackling this

Reply:
I actually had a plan to move this to Alloy , so the dumping logic , decoding can live in one place , to provide reuse for the Disney addition too

Reply:
done